Chap_API_Query.tex

```diff
@@ -0,0 +1,662 @@
+}
+%
+\pasteAttributeItem{PMIX_QUERY_PROC_TABLE}
+%
+\pasteAttributeItem{PMIX_QUERY_LOCAL_PROC_TABLE}
+%
+\declareAttribute{PMIX_QUERY_AUTHORIZATIONS}{"pmix.qry.auths"}{bool}{
+Return operations the \ac{PMIx} tool is authorized to perform.  NO QUALIFIERS.
+}
+%
+\declareAttribute{PMIX_QUERY_SPAWN_SUPPORT}{"pmix.qry.spawn"}{bool}{
+Return a comma-delimited list of supported spawn attributes. NO QUALIFIERS.
+}
+%
+\declareAttribute{PMIX_QUERY_DEBUG_SUPPORT}{"pmix.qry.debug"}{bool}{
+Return a comma-delimited list of supported debug attributes. NO QUALIFIERS.
+}
+%
+\declareAttribute{PMIX_QUERY_MEMORY_USAGE}{"pmix.qry.mem"}{bool}{
+Return information on memory usage for the processes indicated in the qualifiers. OPTIONAL QUALIFIERS: \refattr{PMIX_NSPACE} and \refattr{PMIX_RANK}, or \refattr{PMIX_PROCID} of specific process(es) whose memory usage is being requested.
+}
+%
+\declareAttribute{PMIX_TIME_REMAINING}{"pmix.time.remaining"}{char*}{
+Query number of seconds (\code{uint32_t}) remaining in allocation for the specified namespace. OPTIONAL QUALIFIERS: \refattr{PMIX_NSPACE} of the namespace whose info is being requested (defaults to allocation containing the caller).
+}
+%
+\declareAttributeNEW{PMIX_QUERY_ATTRIBUTE_SUPPORT}{"pmix.qry.attrs"}{bool}{
+Query list of supported attributes for specified \acp{API}. REQUIRED QUALIFIERS: one or more of \refattr{PMIX_CLIENT_FUNCTIONS}, \refattr{PMIX_SERVER_FUNCTIONS}, \refattr{PMIX_TOOL_FUNCTIONS}, and \refattr{PMIX_HOST_FUNCTIONS}.
+}
+%
+\pasteAttributeItem{PMIX_QUERY_NUM_PSETS}
+%
+\pasteAttributeItem{PMIX_QUERY_PSET_NAMES}
+%
+\pasteAttributeItem{PMIX_QUERY_PSET_MEMBERSHIP}
+%
+\declareAttributeNEW{PMIX_QUERY_AVAIL_SERVERS}{"pmix.qry.asrvrs"}{pmix_data_array_t*}{
+Return an array of \refstruct{pmix_info_t}, each element itself containing a \refattr{PMIX_SERVER_INFO_ARRAY} entry holding all available data for a server on this node to which the caller might be able to connect.
+}
+%
+\vspace{\baselineskip}
+These keys are used to query memory available and used in the system.
+
+%
+\pasteAttributeItemBegin{PMIX_AVAIL_PHYS_MEMORY}OPTIONAL QUALIFERS: \refattr{PMIX_HOSTNAME} or \refattr{PMIX_NODEID} (defaults to caller's node).
+\pasteAttributeItemEnd
+%
+\declareAttribute{PMIX_DAEMON_MEMORY}{"pmix.dmn.mem"}{float}{
+Megabytes of memory currently used by the \ac{RM} daemon on the node. OPTIONAL QUALIFERS: \refattr{PMIX_HOSTNAME} or \refattr{PMIX_NODEID} (defaults to caller's node).
+}
+%
+\declareAttribute{PMIX_CLIENT_AVG_MEMORY}{"pmix.cl.mem.avg"}{float}{
+Average Megabytes of memory used by client processes on node. OPTIONAL QUALIFERS: \refattr{PMIX_HOSTNAME} or \refattr{PMIX_NODEID} (defaults to caller's node).
+}
+
+\vspace{\baselineskip}
+The following attributes are used as qualifiers in queries regarding attribute support within the \ac{PMIx} implementation and/or the host environment:
```

Comment:
This piece of text needs to be removed, it is replaced by the following opener in 8.1.5.

Revision exception.

Reply:
Upon further investigation, this text was meant to introduce a portion of the attributes starting with PMIX_CLIENT_FUNCTIONS.  When thing were moved around this was accidentally not moved with those.  I moved this back to where it was originally.